src/mkdocs_coverage/plugin.py

```diff
@@ -104,18 +101,25 @@ def on_files(self, files: Files, config: MkDocsConfig, **kwargs: Any) -> Files:
                 coviframe.contentWindow.location.reload();
             }
             </script>
-
             """,
         )
-        page_contents = style + iframe + script
-        files.append(
-            File.generated(
-                config=config,
-                src_uri=page_path + ".md",
-                content=page_contents,
-            ),
-        )
-        return files
+
+        coverage_page_content = iframe + script
+        if not page_content:
+            # hide toc and title for automatically generated coverage pages
+            style = textwrap.dedent(
+                """
+                <style>
+                article h1, article > a, .md-sidebar--secondary {
+                    display: none !important;
+                }
+                </style>
+                """,
+            )
+            return style + coverage_page_content
+        if page_content.__contains__(self.config.coverage_inplace_placeholder):
```

Comment:
`__contains__`? 😅 Why not `in`?

```suggestion
        if self.config.coverage_inplace_placeholder in page_content:
```